regparser/notice/xml.py

```diff
@@ -66,6 +66,78 @@ def _set_date_attr(self, date_type, value):
             value = value.isoformat()
         dates_tag.attrib["eregs-{}-date".format(date_type)] = value
 
+    def derive_rins(self, rins=None):
+        """
+        Modify the XML tree so that it contains meta data for regulation id
+        numbers.
+        The Federal Register API implies that documents can have more than one.
+
+        If we're not given a list, extract the information from the XML.
+
+        The XML we're adding will look something like this::
+
+            <EREGS_RINS>
+                <EREGS_RIN rin="2050-AG65" />
+            </EREGS_RINS>
+
+        :arg list rins: RINs, which should be strings.
+
+        :rtype: list
+        :returns: A list of regulation id numbers.
+        """
+        if not rins:
+            rins = []
+            xml_rins = self.xpath('//RIN')
+            for xml_rin in xml_rins:
+                rin = xml_rin.text.replace("RIN", "").strip()
+                rins.append(rin)
+        rins_el = self.xpath('//EREGS_RINS')
+        if rins_el:
+            rins_el = rins_el[0]
+        else:   # Tag wasn't present; create it
+            rins_el = etree.Element("EREGS_RINS")
+        for rin in rins:
+            rin_el = etree.Element("EREGS_RIN", rin=rin)
+            rins_el.append(rin_el)
+        self.xml.insert(0, rins_el)
+        return rins
+
+    def derive_docket_ids(self, docket_ids=None):
+        """
+        Modify the XML tree so that it contains meta data for docket ids.
+
+        If we're not given a list, extract the information from the XML.
+
+        The XML we're adding will look something like this::
+
+            <EREGS_DOCKET_IDS>
+                <EREGS_DOCKET_ID docket_id="EPA-HQ-SFUND-2010-1086" />
+                <EREGS_DOCKET_ID docket_id="FRL-9925-69-OLEM" />
+            </EREGS_DOCKET_IDS>
+
+        :arg list docket_ids: docket_ids, which should be strings.
+
+        :rtype: list
+        :returns: A list of docket_ids.
+        """
+        if not docket_ids:
+            docket_ids = []
+            xml_did_els = self.xpath('//DEPDOC')
+            for xml_did_el in xml_did_els:
+                did_str = xml_did_el.text.replace("[", "").replace("]", "")
+                docket_ids.extend([_.strip() for _ in did_str.split(";")])
```

Comment:
Two minor things here: You don't need to instantiate the list (i.e. you can remove the square brackets). Unrelated: I'd argue that underscore should be used to indicate "this variable doesn't matter" -- so to use it and then reference it seems bad.